src/Dfc.CourseDirectory.WebV2/Features/NewApprenticeshipProvider/FlowModelInitializer.cs

```diff
@@ -1,36 +1,116 @@
-﻿using System;
-using System.Threading.Tasks;
-using Dfc.CourseDirectory.Core.DataStore.CosmosDb;
+﻿using Dfc.CourseDirectory.Core.DataStore.CosmosDb;
 using Dfc.CourseDirectory.Core.DataStore.CosmosDb.Queries;
+using Dfc.CourseDirectory.Core.DataStore.Sql;
+using Dfc.CourseDirectory.Core.DataStore.Sql.Queries;
+using Dfc.CourseDirectory.Core.Models;
 using Dfc.CourseDirectory.Core.Validation;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading.Tasks;
 
 namespace Dfc.CourseDirectory.WebV2.Features.NewApprenticeshipProvider
 {
     public class FlowModelInitializer
     {
         private readonly ICosmosDbQueryDispatcher _cosmosDbQueryDispatcher;
+        private readonly ISqlQueryDispatcher _sqlDbQueryDispatcher;
+        private readonly IStandardsAndFrameworksCache _iStandardandFromworkCache;
 
-        public FlowModelInitializer(ICosmosDbQueryDispatcher cosmosDbQueryDispatcher)
+        public FlowModelInitializer(ICosmosDbQueryDispatcher cosmosDbQueryDispatcher, ISqlQueryDispatcher sqlDbQueryDispatcher, IStandardsAndFrameworksCache iStandardandFromworkCache)
         {
             _cosmosDbQueryDispatcher = cosmosDbQueryDispatcher;
+            _sqlDbQueryDispatcher = sqlDbQueryDispatcher;
+            _iStandardandFromworkCache = iStandardandFromworkCache;
         }
 
         public async Task<FlowModel> Initialize(Guid providerId)
         {
+            var model = new FlowModel();
             var provider = await _cosmosDbQueryDispatcher.ExecuteQuery(
                 new GetProviderById()
                 {
                     ProviderId = providerId
                 });
 
-            var model = new FlowModel();
 
+            var submission = await _sqlDbQueryDispatcher.ExecuteQuery(new GetLatestApprenticeshipQASubmissionForProvider()
+            {
+                ProviderId = providerId
+            });
+
+            var existingSubmission = submission.Match(_ => null,
+                                                      match => match);
+
+            if (existingSubmission != null)
```

Comment:
`if (!(submission is None))` then `existingSubmission` can go away

Reply:
done